src/kbmod/data_interface.py

```diff
@@ -0,0 +1,150 @@
+import multiprocessing as mp
```

Comment:
I think you can clean out a bunch of the imports that are not used for the file loading, including; multiprocessing, result_list, stats_filter, clustering_filters.